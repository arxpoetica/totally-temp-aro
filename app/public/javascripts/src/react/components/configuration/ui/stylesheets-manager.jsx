import React, { Component } from 'react'
import reduxStore from '../../../../redux-store'
import wrapComponentWithProvider from '../../../common/provider-wrapped-component'
import UiActions from './ui-actions'

export class StylesheetManager extends Component {
  constructor (props) {
    super(props)
    this.state = {
      StyleValues: props.initialConfiguration
    }
  }
  

  render () {
    return <div id='divStylesheetForm'>
     <form>
        <div className='form-group row'>
          {/* The value of the selected configuration type */}
          <label className='col-sm-4 col-form-label'>Value</label>
          <div className='col-sm-8'>
            <textarea className='form-control'
              style={{ height: '350px' }}
              value={this.state.StyleValues}
              onChange={event => this.handleChanges(event.target.value)} />
          </div>
        </div>
        {/* Show an error message if we have one */}
        {
          this.state.errorMessage
            ? <div className='alert alert-danger'>{this.state.errorMessage}</div>
            : null
        }
      </form>
      <button className='btn btn-primary float-right' onClick={() => this.saveStylesheettoServer()}>
        <i className='fa fa-save' />Save settings
      </button>
    </div>
  }
  handleChanges (newValue) {
    this.setState({
      StyleValues:  newValue
    })
  }
  saveStylesheettoServer () {
    if(!this.state.StyleValues){
      this.setState({
               errorMessage: `Please Update some CSS values!!`
             })
    }else{
      this.props.saveStylesheetsToServerAndReload(this.state.StyleValues)
    }
  }
}

StylesheetManager.propTypes = {}
const mapStateToProps = (state) => ({
  initialConfiguration: state.configuration.ui.styleValues
})



const mapDispatchToProps = (dispatch, ownProps) => ({
  saveStylesheetsToServerAndReload: (StyleValues) => dispatch(UiActions.saveStylesheetsToServerAndReload(StyleValues))
})

const ConfigurationEditorComponent = wrapComponentWithProvider(reduxStore, StylesheetManager, mapStateToProps, mapDispatchToProps)
export default ConfigurationEditorComponent
