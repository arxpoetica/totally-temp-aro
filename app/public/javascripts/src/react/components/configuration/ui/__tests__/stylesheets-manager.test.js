/* global test expect jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { StylesheetManager } from '../stylesheets-manager'
import { shallow } from 'enzyme'

const sampleConfiguration = '.test{ background-color:red}'

// -----------------------------------------------------------------------------
test('Default Stylesheet component render', () => {
  const component = renderer.create(
    <StylesheetManager styleValues={sampleConfiguration} />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('should call onChange prop with input value', () => {
  const onChangeMock = jest.fn()
  const component = shallow(<StylesheetManager
    handleChanges={onChangeMock} />)
  component.find('textarea').simulate('change', { target: { value: '.tool-bar {background-color: #00ff00;} ' } })
  expect(component.state('styleValues')).toEqual('.tool-bar {background-color: #00ff00;} ')
})
