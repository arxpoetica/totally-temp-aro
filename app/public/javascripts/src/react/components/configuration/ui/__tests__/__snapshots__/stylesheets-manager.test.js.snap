// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Stylesheet component render 1`] = `
<div
  id="divStylesheetForm"
>
  <form>
    <div
      className="form-group row"
    >
      <label
        className="col-sm-4 col-form-label"
      >
        Value
      </label>
      <div
        className="col-sm-8"
      >
        <textarea
          className="form-control"
          onChange={[Function]}
          style={
            Object {
              "height": "350px",
            }
          }
        />
      </div>
    </div>
  </form>
  <button
    className="btn btn-primary float-right save-stylesheet"
    onClick={[Function]}
  >
    <i
      className="fa fa-save"
    />
    Save settings
  </button>
</div>
`;
